### PR TITLE
feat: add fallback to LRCLIB for fetching lyrics and implement parsin…

### LINKED ISSUE #78: Add lyrics source
### Summary

Add this source for best sync lyrics
https://lrclib.net/

### Basic Example

Please try to add 

### Reference Issues

_No response_

### Unresolved questions

_No response_

### Would you like to work on this feature?

- [ ] I would like to work on this feature

### PR DESCRIPTION
**Pull request by nitinmano**
# Description

This pull request adds **lrclib.net** as a secondary source for lyrics in the BetterLyrics extension. When the primary source (`lyrics-api.boidu.dev`) does not return any lyrics or encounters an error, the application will now attempt to fetch lyrics from **lrclib.net**.

**Changes include:**

- Implemented `fetchFromLrclib` function to fetch lyrics from lrclib.net.
- Added `parseLRCLIBLyrics` function to parse the LRC-formatted lyrics from lrclib.net into the expected format.
- Modified `createLyrics` function to attempt fetching from lrclib.net if the primary source fails or returns no lyrics.
- Ensured that the lyrics are correctly synchronized and displayed in the application.

**Motivation and Context:**

This change addresses user request **[#78](https://github.com/boidushya/better-lyrics/issues/78)**, where a user specifically asked for **lrclib.net** to be added as a lyrics source for better synced lyrics. By adding lrclib.net as a secondary source, we increase the availability and accuracy of lyrics, enhancing the user experience.

**Dependencies:**

No new dependencies are required for this change.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Related Issue (if applicable)

Fixes #78

## Screenshots (if applicable)

| Before                               | After                                |
| ------------------------------------ | ------------------------------------ |
|
Screenshot 2024-10-16 at 11 11 32 PM
 |  Screenshot 2024-10-16 at 11 12 35 PM  |

## Checklist

- [x] I have performed a self-review of my code
- [x] Will this be part of a product update? If yes, please write one phrase about this update.

**Product Update:**

- Added lrclib.net as a secondary lyrics source to improve lyrics availability.

### CODE REVIEW

File: src/modules/lyrics/lyrics.js [hunk 1]
```diff
@@ -5,7 +5,29 @@ BetterLyrics.Lyrics = {
       const artist = e.artist;
 
       BetterLyrics.Utils.log(BetterLyrics.Constants.FETCH_LYRICS_LOG, song, artist);
+      async function fetchFromLrclib(song, artist) {
+        const lrclibUrl = `https://lrclib.net/api/get?track_name=${encodeURIComponent(song)}&artist_name=${encodeURIComponent(artist)}`;
```
Comment (boidushya) [inline src/modules/lyrics/lyrics.js]: Let's move all the constants to constants.js and reference them here for better maintainability like we do for better-lyrics-api

File: src/modules/lyrics/lyrics.js [hunk 2]
```diff
@@ -5,7 +5,29 @@ BetterLyrics.Lyrics = {
...
 
+        const response = await fetch(lrclibUrl, {
+          headers: {
+            'Lrclib-Client': 'BetterLyrics Extension (https://github.com/boidushya/better-lyrics)'
```
Comment (boidushya) [inline src/modules/lyrics/lyrics.js]: same here (the header value)

File: src/modules/lyrics/lyrics.js [hunk 3]
```diff
@@ -5,7 +5,29 @@ BetterLyrics.Lyrics = {
...
+          }
+        });
+
+        if (!response.ok) {
+          throw new Error(`HTTP error! status: ${response.status}`);
```
Comment (boidushya) [inline src/modules/lyrics/lyrics.js]: .

File: src/modules/lyrics/lyrics.js [hunk 4]
```diff
@@ -5,7 +5,29 @@ BetterLyrics.Lyrics = {
...
+        }
+
+        const data = await response.json();
+
+        if (data && data.syncedLyrics) {
+          BetterLyrics.Utils.log('Lyrics found from LRCLIB');
```
Comment (boidushya) [inline src/modules/lyrics/lyrics.js]: .

File: src/modules/lyrics/lyrics.js [hunk 5]
```diff
@@ -5,7 +5,29 @@ BetterLyrics.Lyrics = {
...
+          const parsedLyrics = BetterLyrics.Lyrics.parseLRCLIBLyrics(data.syncedLyrics, data.duration);
+          return parsedLyrics;
+        } else {
+          throw new Error('No lyrics found on LRCLIB');
```
Comment (boidushya) [inline src/modules/lyrics/lyrics.js]: .

**Comment by nitinmano** [comment]
done

**Comment by boidushya** [comment]
thanks! gonna check them tomorrow 🤝

**Comment by boidushya** [comment]
Moved to #81

**Pull request closed without merging**